### ISSUE TITLE
Lollypop does not see anything else other than "music" in the home directory

### ISSUE THREAD

**Issue by stragu** [flathub/org.gnome.Lollypop#123]
(originally reported on [the Lollypop repo](https://gitlab.gnome.org/World/lollypop/-/issues/2661) but Cédric said it is a Flatpak issue)

### Environment

- Lollypop version: 1.4.8
- GTK+ version: 2.24.32 and 3.22.30
- Operating system: Ubuntu 18.04

### Bug

Since the last update (1.4.8), and after a prompt about re-adding the library folders, Lollypop does not see my music library in `~/Musique`, and doesn't see anything else than the `~/Music` folder in my home, so I can't pick the right directory again.

```
flatpak run org.gnome.Lollypop --debug
Gtk-Message: 14:27:13.499: Failed to load module "canberra-gtk-module"
[INFO] 2021-01-08 14:27:13 Collection web service started
[INFO] 2021-01-08 14:27:13 Collection download started
[INFO] 2021-01-08 14:27:13 Collection download finished
[INFO] 2021-01-08 14:27:13 Last.fm web service started
[INFO] 2021-01-08 14:27:13 ListenBrainz web service started
[INFO] 2021-01-08 14:27:13 Scan started
[INFO] 2021-01-08 14:27:14 lollypop.collection_scanner::__get_objects_for_uris: execution time 0:0.005454
[DEBUG] 2021-01-08 14:27:14 Importing audio file:///home/stragu/Music/Mixxx/Recordings/2019-02-04_23h17m33s.flac
[DEBUG] 2021-01-08 14:27:14 Importing audio file:///home/stragu/Music/Mixxx/Recordings/2019-03-04_21h56m34s.flac
[DEBUG] 2021-01-08 14:27:14 Importing audio file:///home/stragu/Music/Mixxx/Recordings/2018-12-15_14h19m49s.flac
[ERROR] 2021-01-08 14:27:14 AlbumArtwork::__get_pixbuf_from_tags(): gst-resource-error-quark: Resource not found. (3)
[WARNING] 2021-01-08 14:27:14 AlbumArtwork::get(): None -> g-io-error-quark: Error opening directory '/home/stragu/Musique/El Guincho/(2007) Alegranza': No such file or directory (1)
[DEBUG] 2021-01-08 14:27:14 LazyLoadingView::lazy_loading(): 0.7348673343658447
[WARNING] 2021-01-08 14:27:14 Removed, not in collection anymore: file:///home/stragu/Musique/Deru/(2003)%20Pushing%20Air/10%20Timebox.flac -> ['file:///home/stragu/Music']
[WARNING] 2021-01-08 14:27:14 Removed, not in collection anymore: file:///home/stragu/Musique/Deru/(2003)%20Pushing%20Air/14%20Flux%20of%20Humour.flac -> ['file:///home/stragu/Music']
[WARNING] 2021-01-08 14:27:14 Removed, not in collection anymore: file:///home/stragu/Musique/Deru/(2003)%20Pushing%20Air/04%20Echos%20of%20Me.flac -> ['file:///home/stragu/Music']
[...]
```

... and it keeps going for the rest of the collection.

When trying to add the right directory:

![image](https://user-images.githubusercontent.com/1747497/104121906-8ada7980-538d-11eb-9449-31e8cc89776d.png)

**Comment by Eonfge** [comment]
I have a few suggestions.

- Can you try the following: `rm ~/.local/share/flatpak/overrides/org.gnome.Lollypop` This removes all possible custom permissions that you have.

- Can you check for me the folder configuration, this might be a localization issue. Afterall, you refer to the folder `./Musique` but your desktop might look for `./Music` due to a localization issue. Please, give me the results of `cat ~/.config/user-dirs.dirs` and `cat ~/.config/user-dirs.locale`

- Third option: Can you go to Preferences -> Music and show me a screenshot of what is listed there? Perhaps you should manually add `./Musique` because the computer is configurated to look for `./Music`

**Comment by Jordanwvn** [comment]
I have had the same issue, with the same environment, except I was previously using a ./Music directory on an external hard drive. After the last update, Lollypop is no longer able to find any content on my external drive so I cannot select the correct directory.

**Comment by Eonfge** [comment]
> I had the same issue, with the same environment, except I was previously using a `./Music` directory on an external hard drive. It looks like the last update explicitly changes the filesystem access to only allow the local Music directory, as defined by XDG.
> 
> The fix I used was to change `XDG_MUSIC_DIR` in `~/.config/user-dirs.dirs` to instead point to the directory on my external drive. This works for me, because I don't intend to use my local `~/Music` directory at all.

I have the same setup on my own. I have `~./Music` which I leave there for testing purposes, but all my actual music is on `/mnt/data/Muziek`. What I did after the migration, is I went into the preferences menu, tab music, and I added the `/mnt/data/Muziek` directory.

@Jordanwvn Can you tell me how and where you mounted the external disk? Lollypop should support gvfs so almost all situations should be supported.

**Comment by stragu** [comment]
Thanks @Eonfge 

Replying to your three dot points:

* The first command does not work as the file does not exist (i.e. I assume it means I have no custom permissions)
* I use a EN locale:

```
cat ~/.config/user-dirs.dirs
# This file is written by xdg-user-dirs-update
# If you want to change or add directories, just edit the line you're
# interested in. All local changes will be retained on the next run.
# Format is XDG_xxx_DIR="$HOME/yyy", where yyy is a shell-escaped
# homedir-relative path, or XDG_xxx_DIR="/yyy", where /yyy is an
# absolute path. No other format is supported.
# 
XDG_DESKTOP_DIR="$HOME/Desktop"
XDG_DOWNLOAD_DIR="$HOME/Downloads"
XDG_TEMPLATES_DIR="$HOME/Templates"
XDG_PUBLICSHARE_DIR="$HOME/Public"
XDG_DOCUMENTS_DIR="$HOME/Documents"
XDG_MUSIC_DIR="$HOME/Music"
XDG_PICTURES_DIR="$HOME/Pictures"
XDG_VIDEOS_DIR="$HOME/Videos"

cat ~/.config/user-dirs.locale
en_US(base)
```

* The issue is that when I add a source, the file explorer does not show _anything else_ than `~/Music` in my home directory. The folder exists, but it doesn't contain my music library. It's like the application only has the rights to see the `~/Music` directory (see screenshot above). Even clicking on "Desktop" makes it "pop" back to Home.

**Comment by Jordanwvn** [comment]
@Eonfge the external drive I keep my music on is just a portable SSD. So when it's connected to my Ubuntu laptop the music directory is located at `/media/jordan/T7/Music`.

**Comment by Eonfge** [comment]
@stragu Could you try something for me? 
- ` flatpak run --command=sh --devel org.gnome.Lollypop`
- `/app/bin/lollypop`
- Go to Preferences -> Music -> Add Folder (+)

Tell me what logging appeared in your console.

Also, can you run `flatpak --version` because perhaps something is wrong with the Flatpak version you're using.

**Comment by stragu** [comment]
@Eonfge 

I get an error:

```
flatpak run --command=sh --devel org.gnome.Lollypop
error: runtime/org.gnome.Sdk/x86_64/3.38 not installed
```

But I can do:

```
flatpak run org.gnome.Lollypop --debug
[...]
(lollypop:2): Gtk-WARNING **: 21:43:01.725: GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER is not supported by GtkFileChooserNativePortal because portal is too old
```

,,, which might explain something!

And the version:

```
flatpak --version
Flatpak 1.0.9
```

**Comment by Eonfge** [comment]
Aaaah. That's the issue. You're not using the official PPA repository on Ubuntu 18.04 You'll have to add it to your system for newer Flatpak support:
```
$ sudo add-apt-repository ppa:alexlarsson/flatpak
$ sudo apt update
$ sudo apt install flatpak
```
https://www.flatpak.org/setup/Ubuntu/

Alternatively, update to 20.04 or 20.10

**Comment by stragu** [comment]
Thank you! Yes, using the PPA updated flatpak, and I can now see all my files. Still can't add that particular folder for some reason, but closing this for now, unless @Jordanwvn thinks it should stay open?

**Comment by RedMonkeyGone** [comment]
Actually, I think it should stay open. Am having the same issue but on CentOS8. Flatpak ver. 1.6.2

**Comment by Eonfge** [comment]
Yes, and I'm afraid that 1.6 is no longer maintained. 1.8 and 1.10 are maintained, receiving security patches and such.

What you could do as a workaround, is giving lollypop full access rights:
```flatpak override --user org.gnome.Lollypop --filesystem=host```

Keep in mind that this has security implications. You're essentially disabling the sandboxing of Lollypop.

**Comment by RedMonkeyGone** [comment]
Ahh, I see. 
So, the only workaround is to install lollypop independently of flatpak or for the centos repo to keep more up to date with flatpak.

**Comment by Eonfge** [comment]
You might still be better off using Flatpak, disabling the sandbox permissions. In exchange, you have the latest dependencies, and recent updates. Cent OS comes with Lollypop 1.1 so I would recommend using the Flatpak version, even if the sandboxing is not suitable.

**Comment by RedMonkeyGone** [comment]
Thanks, hombre. It works now.

**Comment by linuxitomio** [comment]
I might say that I have Ubuntu 20.04 and the release was not update. It was kept in Lollypop 1.2.2

**Comment by Eonfge** [comment]
@linuxitomio & others. I'm taking this issue upstream because the website doesn't make it terribly clear what to do as an Ubuntu user. See: https://github.com/flatpak/flatpak.github.io/issues/450 for details.

**Tldr: If you use Ubuntu, you must use the PPA.** All versions tend to get outdated pretty quickly otherwise.

```
$ sudo add-apt-repository ppa:alexlarsson/flatpak
$ sudo apt update
$ sudo apt install flatpak
```

**Comment by linuxitomio** [comment]
Thanks for the news.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by linuxitomio** [comment]
and another thing. what about if I am using Debian 10?

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by Eonfge** [comment]
I'm closing this issue, moving the annotations to the README.md using #134
